region/brinstar/pink.json

```diff
@@ -3181,6 +3186,37 @@
                     "Its acceleration will cause it to clip into the blue gate off-screen."
                   ],
                   "devNote": "The reverse gate glitch pretty much requires killing the enemies first, so it's required here."
+                },
+                {
+                  "name": "G-Mode Indirect to Despawn Gate",
+                  "notable": false,
+                  "requires": [
+                    {"comeInWithGMode": {
```

Comment:
A strat for avoiding hoppers should be added.  It would help save health for this GMode strat.
`canPrepareForNextRoom` with consecutivewalljump, spacejump, precise grapple, or speed+runway.

Also this strat needs order priority.  And a 1->1 clear A would happen after since most of those take damage.

Reply:
This strat will use the `Base` 1->2 afterwards. Or it will do another 1->1 that clears the enemies (`A`) after and then get up with speed.

Reply:
Ah, I see now. That does seem like a large problem. I'll work on it.

Reply:
Talking to Maddo, this sounds like it won't actually be a problem. In order to use a `comeInWithGMode` it will require all of the obstacles to not be cleared.

Reply:
Right, here's the relevant code in case anyone is curious: https://github.com/blkerby/MapRandomizer/blob/857b168c602923e586a4f7996f9e77994ab48b76/rust/src/game_data.rs#L1320. If there's an obstacle already cleared, then there's no possibility of satisfying a comeInWithGMode in that state.

Once we finish reworking the cross-room schema, it will only allow entrance conditions (e.g. comeInWithGMode) to be used in conjunction with a strat that enters the room, which should make this even simpler to think about.